Source/JavaScriptCore/builtins/ArrayPrototype.js

```diff
@@ -231,6 +231,8 @@ function fill(value /* [, start [, end]] */)
         if (final > length)
             final = length;
     }
+    if (@arrayFillFast(array, k, final, value))
+        return array;
     for (; k < final; k++)
         array[k] = value;
     return array;
```

Comment:
I feel like we should implement `Array#fill` totally in C++.

Reply:
fixed